src/ngx_http_lua_socket_tcp.c

```diff
@@ -987,7 +995,26 @@ ngx_http_lua_socket_tcp_connect(lua_State *L)
 
         dd("socket key: %s", lua_tostring(L, -1));
 
-    } else { /* n == 2 */
+    } else if (len < 5 || ngx_strncasecmp(p, (u_char *) "unix:", 5) != 0) {
```

Comment:
move the hit of `unix socket domain` to this `else if` block may be better?

Reply:
done